File: Semestru 3/MAP/lab/lab10/lab10/lab10/repo/IRepository.cs
using lab10.domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace lab10.repo
{
	internal interface IRepository<E, ID> where E : Entity<ID>
	{
		public E? Find(ID id);
		public E? Save(E Entity);
		public void Delete(ID id);
		public IEnumerable<E> FindAll();
		public E? Update (E Entity);
	}
}

File: Semestru 4/MPP/lab/java/lab3/Persistence/src/main/java/org/example/repository/interf/IRepository.java
package org.example.repository.interf;

import org.example.domain.Entity;

public interface IRepository<ID, E extends Entity<ID>> {
    E FindById (ID id);
    Iterable<E> FindAll();
    E Save (E entity);
    E Update (E entity);
    E Delete (ID id);
}
